### PR TITLE
Bug 1748452: data/bootstrap: Replace openshift.sh with cluster-bootstrap

### PR DESCRIPTION
**Pull request by wking**
With this commit, I take advantage of openshift/cluster-bootstrap@fc5e0941 (openshift/cluster-bootstrap#14) to replace our previous openshift.sh (with a minor change to the manifest directory).  I'm currently using a cp in bootkube.sh to shift those manifests into the generic directory; I plan on consolidating `Openshift` into `Manifests` in `pkg/asset/manifests` in follow-up work.

This spins off part of my earlier #1147 (hopefully the less-controversial part ;).

### CODE REVIEW

**Comment by wking** [comment]
/hold

We probably want this in 4.1, but if I change the base to point at `master-4.1` the diff will be huge ;).  I'm just going to put a hold on this while it's under review, and if we get to the point where we want to merge it I'll file a separate PR to fast-forward `master-4.1` and then rebase this one on top.

**Comment by wking** [comment]
Rebased with f0dcefe3d -> d60789682.  And since the pivot to loopback kubeconfigs in #2086, having a separate `openshift.sh` pushing in parallel (and possibly after) cluster-bootstrap is now a racy bug.

/retitle Bug 1748452: data/bootstrap: Replace openshift.sh with cluster-bootstrap

**Comment by abhinavdahiya** [comment]
the apiserver team that owns the cluster-bootstrap has made it very clear in the past that it should only be used for bootstrapping the k8s control plane, therefore we have the openshift.service that keeps the separation of concerns, Therefore I don't agree with using the bootkube.service's cluster-bootstrap to push these objects that are not required to boot the k8s control plane.

**Comment by wking** [comment]
[e2e-aws][1] failed with:


```
Sep 10 19:41:16 ip-10-0-3-146 bootkube.sh[1605]: "99_openshift-cluster-api_master-user-data-secret.yaml": unable to convert asset "99_openshift-cluster-api_master-user-data-secret.yaml" to unstructed
Sep 10 19:41:16 ip-10-0-3-146 bootkube.sh[1605]: "99_openshift-cluster-api_worker-user-data-secret.yaml": unable to convert asset "99_openshift-cluster-api_worker-user-data-secret.yaml" to unstructed
```

in `./bundle/bootstrap/journals/bootkube.log` in [the gathered logs][2].  I'm not clear on why yet.

> the apiserver team that owns the cluster-bootstrap has made it very clear in the past that it should only be used for bootstrapping the k8s control plane

I'm not clear yet on why they'd care about them pushing these manifests vs. us pushing these in parallel.  Maybe it's the unstructured casting issue I'm hitting :p.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/1381/pull-ci-openshift-installer-master-e2e-aws/7585
[2]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1381/pull-ci-openshift-installer-master-e2e-aws/7585/artifacts/e2e-aws/installer/log-bundle-20190910200623.tar

**Comment by wking** [comment]
Ah, looks like 9e4b0d484ffb didn't help.  Still getting:

```
Sep 11 19:54:03 ip-10-0-5-50 bootkube.sh[1606]: "99_openshift-cluster-api_master-user-data-secret.yaml": unable to convert asset "99_openshift-cluster-api_master-user-data-secret.yaml" to unstructed
Sep 11 19:54:03 ip-10-0-5-50 bootkube.sh[1606]: "99_openshift-cluster-api_worker-user-data-secret.yaml": unable to convert asset "99_openshift-cluster-api_worker-user-data-secret.yaml" to unstructed
```

from [the e2e-aws log bundle][1].

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1381/pull-ci-openshift-installer-master-e2e-aws/7596/artifacts/e2e-aws/installer/log-bundle-20190911201735.tar

**Comment by wking** [comment]
All green :)

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 1]
```diff
@@ -48,6 +48,13 @@ systemctl restart cri-o.service
 
 mkdir --parents ./{bootstrap-manifests,manifests}
 
+if [ ! -f openshift-manifests.done ]
+then
+	echo "Moving OpenShift manifests in with the rest of them"
+	cp openshift/* manifests/
```
Comment (sttts) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: should be fine

**Comment by abhinavdahiya** [comment]
Also can we drop `dropping report-progess` service, I still think the seprate service for it is useful.

**Comment by wking** [comment]
> Also can we drop `dropping report-progess` service...

Rerolled to keep it with 276840df3 -> 953364203.

**Comment by wking** [comment]
I've pushed 953364203 -> 13e7a4d79 fixing some issues with my `progress.service` [restoration][1] to address the [hung CI install][2].

[1]: https://github.com/openshift/installer/pull/1381#issuecomment-532311622
[2]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/1381/pull-ci-openshift-installer-master-e2e-aws/7638

**Comment by wking** [comment]
GCP [timed out waiting for bootstrap completion][1], despite:

```
Sep 17 19:52:02 ci-op--nn2zb-b.c.openshift-gce-devel-ci.internal bootkube.sh[1754]: bootkube.service complete
```

in `./bootstrap/journals/bootkube.log` in [the gathered logs][2].  Maybe my attempt to restore `progress.service` is still broken...

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/1381/pull-ci-openshift-installer-master-e2e-gcp/59
[2]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1381/pull-ci-openshift-installer-master-e2e-gcp/59/artifacts/e2e-gcp/installer/log-bundle-20190917201747.tar

**Comment by wking** [comment]
Ah, fixed another part of `progress.service` restoration with 13e7a4d79 -> b982886ea.

**Comment by wking** [comment]
All green :).

**Comment by wking** [comment]
Actually, looks like we may need to trigger these manually after a force-push?

/test e2e-gcp
/test e2e-azure

**Comment by wking** [comment]
All green :).  Another round for luck:

/test e2e-aws
/test e2e-azure
/test e2e-gcp

**Comment by wking** [comment]
[Azure failed][1] with:

```
Sep 18 03:15:12.412: INFO: Waiting for the OAuth server route to be ready: EOF
...
fail [github.com/openshift/origin/test/extended/oauth/expiration.go:30]: Unexpected error:
    <*errors.errorString | 0xc000273390>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred
...
failed: (1m39s) 2019-09-18T03:15:44 "[Feature:OAuthServer] [Token Expiration] Using a OAuth client with a non-default token max age to generate tokens that do not expire works as expected when using a code authorization flow [Suite:openshift/conformance/parallel]"
```

We hit that flake [sometimes][2] (although [not usually on Azure][3]).  Doesn't seem like a setup-time issue, but just in case, we'll go around again:

/test e2e-azure

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/1381/pull-ci-openshift-installer-master-e2e-azure/184
[2]: https://ci-search-ci-search-next.svc.ci.openshift.org/?search=failed:%20.*Using%20a%20OAuth%20client%20with%20a%20non-default%20token%20max%20age%20to%20generate%20tokens%20that%20do%20not%20expire%20works%20as%20expected%20when%20using%20a%20code%20authorization%20flow
[3]: https://ci-search-ci-search-next.svc.ci.openshift.org/?name=azure&search=failed:%20.*Using%20a%20OAuth%20client%20with%20a%20non-default%20token%20max%20age%20to%20generate%20tokens%20that%20do%20not%20expire%20works%20as%20expected%20when%20using%20a%20code%20authorization%20flow

**Comment by wking** [comment]
All green again :)

**Comment by wking** [comment]
@abhinavdahiya, green enough?

**Pull request merged**